packages/core/src/tech/HlsJsTech.ts

```diff
@@ -75,6 +75,39 @@ export default class HlsJsTech extends BaseTech {
     this.playlistDuration = data?.details?.totalduration;
   }
 
+  getVideoQualities(){
+    if(this.hls){
+      const videoQualities: IVideoQuality[] = [];
+      const levels = this.hls.levels;
+      levels.forEach(level => {
+        const quality: IVideoQuality = {
+          width: level['width'],
+          height: level['height'],
+          bitrate: level['bitrate'],
+        }
+        videoQualities.push(quality);
+      });
+      return videoQualities;
+    }
+  }
+
+  get currentLevel(){
+    if(this.hls)
+      return this.hls.currentLevel;
+    return NaN;
```

Comment:
`NaN` is generally considered bad practise, the function will automatically return `undefined` without it which is fine.

Reply:
yessir